test/units/parsing/test_splitter.py

```diff
@@ -173,3 +177,9 @@ def test_split_args_error(args):
 def test_parse_kv_error(args):
     with pytest.raises(AnsibleParserError):
         parse_kv(args)
+
+
+@pytest.mark.parametrize("args", REMAINDER_ERROR)
+def test_parse_kv_remainder_error(args):
```

Comment:
Include type hints on new code, including unit tests, when reasonable:

```suggestion
def test_parse_kv_remainder_error(args: str) -> None:
```